src/mods/vr/runtimes/OpenVR.cpp

```diff
@@ -194,12 +209,55 @@ VRRuntime::Error OpenVR::update_matrices(float nearz, float farz){
 
     this->hmd->GetProjectionRaw(vr::Eye_Left, &this->raw_projections[vr::Eye_Left][0], &this->raw_projections[vr::Eye_Left][1], &this->raw_projections[vr::Eye_Left][2], &this->raw_projections[vr::Eye_Left][3]);
     this->hmd->GetProjectionRaw(vr::Eye_Right, &this->raw_projections[vr::Eye_Right][0], &this->raw_projections[vr::Eye_Right][1], &this->raw_projections[vr::Eye_Right][2], &this->raw_projections[vr::Eye_Right][3]);
+    // SPDLOG_INFO("Original left  {}, {}, {}, {}", this->raw_projections[0][0], this->raw_projections[0][1],this->raw_projections[0][2], this->raw_projections[0][3]);
+    // SPDLOG_INFO("Original right {}, {}, {}, {}", this->raw_projections[1][0], this->raw_projections[1][1],this->raw_projections[1][2], this->raw_projections[1][3]);
 
+    auto& vr = VR::get();
     auto get_mat = [&](vr::EVREye eye) {
-        const auto left =   this->raw_projections[eye][0] * -1.0f;
-        const auto right =  this->raw_projections[eye][1] * -1.0f;
-        const auto top =    this->raw_projections[eye][2] * -1.0f;
-        const auto bottom = this->raw_projections[eye][3] * -1.0f;
+        const auto tan_half_fov = new float[4];
```

Comment:
Very big alarm bells when I saw this. Do NOT call `new`, we do not allocate raw pointers in modern C++ without intention, we prefer `std::unique_ptr` or `std::shared_ptr`. We also do not call `new` without calling `delete`. This is creating a memory leak every time this is called.

Change this to `std::array<float, 4> tan_half_fov{};`.

Reply:
Yikes, I did this after a day in the office cutting Java and relying on garbage collection. Thanks for checking, will rework it and take better care

Reply:
changed as suggested, feel suitably embarrassed